app/viewmodels/DutyDefermentStatementsForEori.scala

```diff
@@ -43,8 +50,13 @@ case class DutyDefermentStatementsForEori(eoriHistory: EoriHistory,
     }.toSeq.sorted
   }
 
-  private def groupByMonthAndYear(periods: Seq[DutyDefermentStatementPeriod]): Seq[DutyDefermentStatementPeriodsByMonth] =
-  {
+  private def filterDates(startDate: LocalDate,
+                          endDate: LocalDate,
+                          periods: Seq[DutyDefermentStatementPeriodsByMonth]): Seq[DutyDefermentStatementPeriodsByMonth] = {
+    periods.filter(dds => Utils.isEqualOrAfter(dds.monthAndYear, startDate) && Utils.isEqualOrBefore(dds.monthAndYear, endDate))
```

Comment:
Better to remove Utils to access its methods. Import it and then use it without Utils as it will improve the readability and if someone need to look into these method, they can click the method and can have a look.